api/services/gameService.js

```diff
@@ -106,62 +96,39 @@ module.exports = {
    ** Return a fully populated Game as a Promise
    ****options = {gameId: gameId}
    */
-  populateGame: function (options) {
-    return new Promise(function (resolve, reject) {
-      if (options) {
-        if (Object.hasOwnProperty.call(options, 'gameId') && typeof options.gameId === 'number') {
-          // find game
-          return (
-            gameService
-              .findGame({ gameId: options.gameId })
-              // then find users
-              .then(function findUsers(game) {
-                if (game.players) {
-                  if (game.players.length > 1) {
-                    const p0 = userService.findUser({ userId: game.players[0].id });
-                    const p1 = userService.findUser({ userId: game.players[1].id });
-                    const spectators = fetchSpectatorUsernames(game.id);
-                    return Promise.all([Promise.resolve(game), p0, p1, spectators]);
-                  }
-                  return Promise.reject({ message: "Can't populate game without two players" });
-                }
-                return Promise.reject({
-                  message: "Can't populate game, because it does not have players collection",
-                });
-              })
-              // then find points
-              .then(function findPoints(values) {
-                const [game, p0, p1, spectators] = values;
-                const p0Points = cardService.findPoints({ userId: p0.id });
-                const p1Points = cardService.findPoints({ userId: p1.id });
-                return Promise.all([
-                  Promise.resolve(game),
-                  Promise.resolve(p0),
-                  Promise.resolve(p1),
-                  Promise.resolve(spectators),
-                  p0Points,
-                  p1Points,
-                ]);
-              })
-              // then format results & resolve
-              .then(function finish(values) {
-                const [game, p0, p1, spectators, p0Points, p1Points] = values;
-                const populatedP0 = new tempUser(p0, p0Points);
-                const populatedP1 = new tempUser(p1, p1Points);
-                const result = new tempGame(game, spectators, populatedP0, populatedP1);
-                return resolve(result);
-              })
-              .catch(function failed(err) {
-                reject(err);
-              })
-          );
+   populateGame: async function(options) {
+    try{
+      if (!options) {
+        if (!Object.hasOwnProperty.call(options, 'gameId') && !typeof options.gameId === 'number') {
+          throw new Error({message: 'gameId is required and must be a number'});
         }
-        reject({ message: 'gameId is required and must be a number' });
-      } else {
-        reject({ message: 'Cannot populate Game without GameId (options had no gameId)' });
+        throw new Error({ message: 'Cannot populate Game without GameId (options had no gameId)' }); 
```

Comment:
So this sortof highlights the awkwardness of using an object param when the only data needed is a single integer. It would probably be better if this were just `gameId` instead of `options`. That would require updating every instance where we call `populateGame()` to just pass the gameId directly without wrapping it in an object, but then  the validation here would be a simpler single call to `Number.isIngeger(gameId)`

What do you think?

Reply:
Relatedly, as documented in #188, we want to completely remove all the services and hooks to replace them with helpers, since all three serve the same purpose and helpers is the newer approach. So if we *do* opt to update the param to be a single int instead of an object and therefore update every call to populateGame(), it might make sense to go a little further and rewrite this function as a helper. Granted that increases the scope of this PR, so it might make sense to push both of those changes into a separate issue. What do you think?

Reply:
I like the idea of removing the object param, I think it will close off one extra step that people may miss. I think we should handle this is #188 since we're going to have to go back to every .populateGame to do that issue anyways. I'm happy to pick up that issue too 👨‍💻